home-manager/home-manager

```diff
@@ -67,18 +66,55 @@ function doBuild() {
         extraArgs="$extraArgs --show-trace"
     fi
 
-    nix-build $extraArgs \
-              "@HOME_MANAGER_EXPR_PATH@" \
-              --argstr confPath "$HOME_MANAGER_CONFIG" \
-              --argstr confAttr "$HOME_MANAGER_CONFIG_ATTRIBUTE" \
-              -A activationPackage
+    nix-build \
+        "@HOME_MANAGER_EXPR_PATH@" \
+        $extraArgs \
+        --argstr confPath "$HOME_MANAGER_CONFIG" \
+        --argstr confAttr "$HOME_MANAGER_CONFIG_ATTRIBUTE"
+}
+
+function presentNews() {
+    local infoFile
+    infoFile=$(doBuildNews -A newsInfo) || return 1
+
+    # shellcheck source=/dev/null
+    . "$infoFile"
+
+    if [[ $newsNumUnread -eq 0 ]]; then
+        return
+    elif [[ "$newsDisplay" == "notify" ]]; then
+        # Not actually an error but here stdout is reserved for
+        # nix-build output.
+        errorEcho
+        if [[ $newsNumUnread -eq 1 ]]; then
+            errorEcho "There is an unread and relevant news item."
+            errorEcho "Read it by running the command '$(basename "$0") news'."
+        else
+            errorEcho "There are $newsNumUnread unread and relevant news items."
+            errorEcho "Read them by running the command '$(basename "$0") news'."
+        fi
+        errorEcho
+    elif [[ "$newsDisplay" == "show" ]]; then
+        doShowNews --unread
+    else
+        errorEcho "Unknown 'news.display' setting '$newsDisplay'."
+    fi
+}
+
+function doBuild() {
+    presentNews
```

Comment:
I'd present the news after the build.
1. In this way, news are only shown when the build succeeds.
2. You're more likely to miss the news when it's followed by build output, especially in verbose mode.

Reply:
Fair enough, I think that makes more sense. I'll move this to the bottom.